src/main/kotlin/org/partiql/scribe/targets/spark/SparkCalls.kt

```diff
@@ -1,6 +1,255 @@
 package org.partiql.scribe.targets.spark
 
+import org.partiql.ast.Ast.exprCall
+import org.partiql.ast.Ast.exprCast
+import org.partiql.ast.Ast.exprLit
+import org.partiql.ast.Ast.exprOperator
+import org.partiql.ast.DataType
+import org.partiql.ast.DatetimeField
+import org.partiql.ast.Identifier
+import org.partiql.ast.Literal
+import org.partiql.ast.expr.Expr
 import org.partiql.scribe.ScribeContext
+import org.partiql.scribe.problems.ScribeProblem
+import org.partiql.scribe.sql.SqlArg
+import org.partiql.scribe.sql.SqlArgs
+import org.partiql.scribe.sql.SqlCallFn
 import org.partiql.scribe.sql.SqlCalls
 
-public class SparkCalls(context: ScribeContext) : SqlCalls(context)
+public open class SparkCalls(context: ScribeContext) : SqlCalls(context) {
+    private val listener = context.getProblemListener()
+
+    override val rules: Map<String, SqlCallFn> =
+        super.rules.toMutableMap().apply {
+            this["utcnow"] = ::utcnow
+            this["current_user"] = ::currentUser
+        }
+
+    private fun currentUser(args: List<SqlArg>): Expr {
+        val currentUser = Identifier.regular("current_user")
+        listener.report(
+            ScribeProblem.simpleInfo(
+                ScribeProblem.TRANSLATION_INFO,
+                "PartiQL CURRENT_USER was replaced by Spark `current_user()`",
+            ),
+        )
+        return exprCall(currentUser, emptyList())
+    }
+
+    // convert_timezone('UTC', current_timestamp())
+    // the default return of current_timestamp function is session time zone dependent.
+    // current_timestamp() : https://spark.apache.org/docs/latest/api/sql/index.html#current_timestamp
+    // convert_timezone() : https://spark.apache.org/docs/latest/api/sql/index.html#convert_timezone
+    private fun utcnow(args: List<SqlArg>): Expr {
+        val convertTimeZone = Identifier.regular("convert_timezone")
+        listener.report(
+            ScribeProblem.simpleInfo(
+                ScribeProblem.TRANSLATION_INFO,
+                "PartiQL `utcnow()` was replaced by Spark `convert_timezone('UTC', current_timestamp())`",
+            ),
+        )
+        val currentTimestamp = Identifier.regular("current_timestamp")
+        val targetTimezone = exprLit(Literal.string("UTC"))
+        return exprCall(convertTimeZone, listOf(targetTimezone, exprCall(currentTimestamp, emptyList())))
+    }
+
+    /**
+     * PartiQL: date_add(part: datetime_part, quantity: int, date: date|timestamp) -> date|timestamp`
+     * Spark:   date + interval
+     *
+     * We perform the conversion by a
```

Comment:
`We perform the conversion by a` missing the rest of the sentence?

Reply:
Thanks, that was missing from the original Scribe translation.